### ISSUE TITLE
How to fed 1d input to a convolution1d layer and then pass its output to a maxpooling layer?

### ISSUE THREAD

**Issue by wailoktam** [fchollet/keras#2824]
Hi, I get the following executable code with the maxpooling commented out:

   test3dLArray = numpy.random.random((1, 10,100))
   test3dRArray = numpy.random.random((1, 10,100))
   testLabels = numpy.random.randint(2, size=1)
   testLabels = np_utils.to_categorical(testLabels, 2)

   leftKerasModel = Sequential()
   leftKerasModel.add(Reshape((1000,), input_shape=(10,100)))
   leftKerasModel.add(Dense(200))
   leftKerasModel.add(Reshape((1, 200)))
   convolution1d_1 = Convolution1D(10, 3, border_mode='same')
   leftKerasModel.add(convolution1d_1)
   leftKerasModel.add(Activation('relu'))
   #   leftKerasModel.add(MaxPooling1D(pool_length=2, stride=None, border_mode='valid'))
   leftKerasModel.add(Flatten())

   rightKerasModel = Sequential()
   rightKerasModel.add(Reshape((1000,), input_shape=(10,100)))
   rightKerasModel.add(Dense(200))
   rightKerasModel.add(Reshape((1, 200)))
   convolution1d_2 = Convolution1D(10, 3, border_mode='same')
   rightKerasModel.add(convolution1d_2)
   rightKerasModel.add(Activation('relu'))
   #  rightKerasModel.add(MaxPooling1D(pool_length=2, stride=None, border_mode='valid'))
   rightKerasModel.add(Flatten())

   mergeLayer = Sequential()
   mergeLayer.add(Merge([leftKerasModel, rightKerasModel], mode='cos', dot_axes=1))
   mergeLayer.add(Lambda(lambda x: 1-x))
   mergeLayer.compile(loss='mse', optimizer='sgd')
   mergeLayer.fit([test3dLArray, test3dRArray], testLabels, nb_epoch=10, batch_size=32)

But I get complaints from Keras about handling a dimension assigned a value 0 after I put back the max pooling line. I suspect the convolution1d is not applied in a proper way to its 1,200 input (I intend a 1d convolution moving horizontally in the 200 dimension but I have no way to get convolution1d to accept the (200) input unless I add an extra channel dimension and assign it a value 1) and so the max pooling of length 2 has troubled with the output of some unintended convolutional operation on the added dimension. 

Thanks in advance for any response.

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [O ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [O ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [O ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
